From 95822aab7b236ea98a985431de7dbeefe3dfe1c8 Mon Sep 17 00:00:00 2001
From: Michael Bestas <mkbestas@lineageos.org>
Date: Sun, 4 Feb 2018 21:27:06 +0200
Subject: [PATCH 8/9] SUW: Fix crash on fingerprint capability check

* Check if the device supports the fingerprint feature before proceeding

Change-Id: I9cda52500c1e45499b5647fccff3d7d37c8311ef
---
 src/org/lineageos/setupwizard/util/SetupWizardUtils.java | 11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

diff --git a/src/org/lineageos/setupwizard/util/SetupWizardUtils.java b/src/org/lineageos/setupwizard/util/SetupWizardUtils.java
index 519294e..85f8cb5 100644
--- a/src/org/lineageos/setupwizard/util/SetupWizardUtils.java
+++ b/src/org/lineageos/setupwizard/util/SetupWizardUtils.java
@@ -225,9 +225,14 @@ public class SetupWizardUtils {
     }
 
     public static boolean hasFingerprint(Context context) {
-        FingerprintManager fingerprintManager = (FingerprintManager)
-                context.getSystemService(Context.FINGERPRINT_SERVICE);
-        return fingerprintManager.isHardwareDetected();
+        PackageManager packageManager = context.getPackageManager();
+        if (packageManager.hasSystemFeature(PackageManager.FEATURE_FINGERPRINT)) {
+            FingerprintManager fingerprintManager = (FingerprintManager)
+                    context.getSystemService(Context.FINGERPRINT_SERVICE);
+            return fingerprintManager.isHardwareDetected();
+        } else {
+            return false;
+        }
     }
 
     public static boolean simMissing() {
-- 
2.7.4

